OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -402,7 +402,7 @@ private void setDataForRedisplay(OSInAppMessage message) {
     private void queueMessageForDisplay(@NonNull OSInAppMessage message) {
         synchronized (messageDisplayQueue) {
             // Make sure no message is ever added to the queue more than once
-            if (!isInAppMessageQueued(message)) {
```

Comment:
Using contains with an oObject is dangerous. This could be true or false and not represent the `String messageId`, this should be changed back

Reply:
Yeah agree, thas why i have the equals and hash overriden